firmware/lib/widget/widgetSet.cpp

```diff
@@ -56,23 +56,22 @@ void WidgetSet::prev() {
 void WidgetSet::switchWidget() {
   m_screenManager->clearAllScreens();
   getCurrent()->setup();
+  uint32_t start = millis();
   getCurrent()->draw(true);
+  uint32_t end = millis();
+  Serial.printf("Drawing of %s took %d ms\n", getCurrent()->getName().c_str(), (end-start));
```

Comment:
Please note that printing might slow down stuff as well.
I'd suggest to keep statistics per widget, and print those ones every few seconds or so.

Reply:
This is only printed once if you change the widget (by pressing a button or by the automatic widget cycling). 
Therefore, I don't think that will have much impact.
But I can remove it if you want to.